### PR TITLE
DOC-1753: Document the new Spelling Autocorrect plugin

### PR DESCRIPTION
**Pull request by brian-forte**
Related Ticket: 

Description of Changes:
* Document the new Spelling Autocorrect plugin

Pre-checks:
- [x] Branch prefixed with `feature/6/` or `hotfix/6/`
- [x] `modules/ROOT/nav.adoc` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed

### CODE REVIEW

File: modules/ROOT/pages/autocorrect.adoc
Comment (MitchCTiny) [inline modules/ROOT/pages/autocorrect.adoc:66]: Should these be moved to the bottom of the page? I don't think the first thing people want to see is page of issues with our new plugin. Also, sounds like case-sensitivity should be a feature rather than a known issue as I doubt it needs to be "fixed". The conflict is still an issue though.

**Review by MitchCTiny — APPROVED** [review]

File: modules/ROOT/pages/autocorrect.adoc
Comment (brian-forte) [reply modules/ROOT/pages/autocorrect.adoc]: > Should these be moved to the bottom of the page?

Yes.

> Also, sounds like case-sensitivity should be a feature rather than a known issue as I doubt it needs to be "fixed". The conflict is still an issue though.

To paraphrase my commit message, a bit of thinking about this lead me to *re-structure the whole document and make multiple edits and corrections*.

Comment (lnewson) [reply modules/ROOT/pages/autocorrect.adoc]: @brian-forte I assume the move was in relation this, but this is not how things normally are done so doing this would be inconsistent

Comment (lnewson) [reply modules/ROOT/pages/autocorrect.adoc]: At the very least the common/basic usage information should be at the top. As for limitations, as I said it'd be inconsistent so if you're moving one you really should be moving all. For example https://www.tiny.cloud/docs/tinymce/6/export/#client-side-pdf has the basic usage and limitations together at the top just below the example and before the options.

Comment (brian-forte) [reply modules/ROOT/pages/autocorrect.adoc]: > I assume the move was in relation this, but this is not how things normally are done so doing this would be inconsistent

Yep. My bad. Was too quick on the cut-and-paste-and-get-re-writing front. The **Options** and **Events** sections now round out the page.

Comment (lnewson) [reply modules/ROOT/pages/autocorrect.adoc]: That looks better thanks 👍

**Review by ephox-mogran — COMMENTED** [review]
There are a few things that I'd like you to look at before we merge this. I could probably hit Approve without much convincing, though.

File: modules/ROOT/pages/autocorrect.adoc
Comment (ShiridiGandham) [reply modules/ROOT/pages/autocorrect.adoc:82]: How about - "Auto-capitalization interferes with Enhanced Media Embed"

File: modules/ROOT/partials/configuration/autocorrect_service_url.adoc
Comment (lnewson) [reply modules/ROOT/partials/configuration/autocorrect_service_url.adoc:12]: You're correct, this should not have this. See https://www.tiny.cloud/docs/tinymce/6/export/#export_image_proxy_service_url and https://www.tiny.cloud/docs/tinymce/6/introduction-to-tiny-spellchecker/#spellchecker_rpc_url

File: modules/ROOT/partials/index-pages/premium-plugins.adoc
Comment (lnewson) [reply modules/ROOT/partials/index-pages/premium-plugins.adoc:118]: Capitalisation is grammar 🤷 that said it's not a large part so maybe this is confusing.

File: modules/ROOT/pages/autocorrect.adoc
Comment (lnewson) [reply modules/ROOT/pages/autocorrect.adoc:82]: Yeah using `media embeds` here would be confusing, as that will also mean media inserted via the `media` plugin and this only interferes with Enhanced Media as-you-type functionality IIUC.

File: modules/ROOT/partials/index-pages/premium-plugins.adoc
Comment (ephox-mogran) [reply modules/ROOT/partials/index-pages/premium-plugins.adoc:118]: The statement "capitalisation is grammar" is actually rather contested, but I'll defer to @brian-forte 's undoubted superior knowledge on such a topic. But as you suggest, I think the problem is that if I saw something telling me that it was autocorrecting my grammar mistakes, I would think things like subject-verb agreement (changing "I is" to "I am")

File: modules/ROOT/pages/autocorrect.adoc
Comment (brian-forte) [reply modules/ROOT/pages/autocorrect.adoc:4]: I’ve been thinking along the same lines. I’ve added some keywords and edited the introductory sentence accordingly.

File: modules/ROOT/pages/autocorrect.adoc
Comment (brian-forte) [reply modules/ROOT/pages/autocorrect.adoc:49]: > The hyphenation here is a llttle bit inconsistent, but I'm not sure if it matters. We have:
> 
> autocorrect, auto-corrected, auto-correct, auto-corrects

Good catch. Doing some quick digging, _autocorrect_ emerged entirely from technical en-US. Specifically from folks producing software akin to this plugin.

And, as is somewhat common when en-US speakers coin new words by compounding, it never took a hyphen. (en-US is notorious for getting rid of hyphens quickly, especially compared to other Englishes, especially en-IN.)

FWIW, the sort-of-jokey-but-perhaps-not-entirely-without-merit hypothesis among linguists is that German speakers had a bigger influence on normative en-US spelling norms than is commonly realised. Compounding in German virtually never uses the hyphen glyph.

Comment (brian-forte) [reply modules/ROOT/pages/autocorrect.adoc:49]: > We also use spelling mistakes in one place, and spelling errors in another. Really minor, I'm just wondering if it's intentional.

Another good catch.

I suspect the proximate source of the inconsistency is the document being put together by _[Dyvers Hands](https://en.wikipedia.org/wiki/Divers_hands)_.

But, on reflection, I think _error_ is the better term here. In common use, _mistake_ has connotations of intent and blame. Someone makes a mistake and things being wrong are, therefore, their fault.

By contrast, _error_ is normally used and read without such connotations. The thing is still in the wrong state, and it’s entirely likely the wrong state is consequent to someone’s action. But you aren’t going to get blamed for an _error_, even if you might also be responsible for correcting it.

Typos, in particular, feel like errors, not mistakes, to me. (And, yes, adding the noun phrase `typographical errors` to the keywords runs counter to this. But `typographical error` is the fixed phrase, although it’s now almost completely replaced by `typo`, even in the most formal of editorial environments.)

File: modules/ROOT/partials/configuration/autocorrect_autocorrect.adoc
Comment (brian-forte) [reply modules/ROOT/partials/configuration/autocorrect_autocorrect.adoc:4]: Indeed. And, as per my earlier comment, I have (now). As per one of my commit messages, I don’t think in en-US and still reflexively think hyphens are useful. It’s a *me* problem.

File: modules/ROOT/partials/configuration/autocorrect_service_url.adoc
Comment (brian-forte) [reply modules/ROOT/partials/configuration/autocorrect_service_url.adoc]: My commit message:

> Removed empty default value line (we don’t put a default value line below the *type* if there is not an actual default value).

File: modules/ROOT/partials/configuration/autocorrect_service_url.adoc
Comment (lnewson) [inline modules/ROOT/partials/configuration/autocorrect_service_url.adoc:11]:
```suggestion
```
Just removing excess spacing.

File: modules/ROOT/partials/index-pages/premium-plugins.adoc
Comment (brian-forte) [reply modules/ROOT/partials/index-pages/premium-plugins.adoc]: > The statement "capitalisation is grammar" is actually rather contested, but I'll defer to @brian-forte 's undoubted superior knowledge on such a topic.

*Must resist urge to engage with this topic at length.* 😃 

> But as you suggest, I think the problem is that if I saw something telling me that it was autocorrecting my grammar mistakes, I would think things like subject-verb agreement (changing "I is" to "I am")

An entirely fair reading. It’s what I would think as well.

Also, on a quick read through all the other short descriptions in this file, this one was [not consistent with most of the others](https://ephocks.atlassian.net/browse/DOC-1807).

So I did a quick re-write (and removed _grammar_ in the process).

File: modules/ROOT/pages/autocorrect.adoc
Comment (brian-forte) [reply modules/ROOT/pages/autocorrect.adoc:73]: Call this a combination of laziness and _I was a SysAdmin for several years_ twitchiness.

1. The laziness
what’s in the docs is a very lightly edited version of what I put down as I was, quite literally, discovering the behaviour while I was going through all the queries directed at me in @ShiridiGandham’s [QA spreadsheet](https://docs.google.com/spreadsheets/d/1YKuFZ-xvUDmFBq4kh96ZiRVy5La2NajVvsbmIuSskNk/edit?usp=sharing). It was quicker to lightly-edit than re-write.
2. The twitchiness
When bugs or known issues are not explicitly and fully documented, I’m immediately brought to mind of way too many times I was at a console at 03:00 trying to work out why my [application others are relying on here] wasn’t working and the notes accompanying the [will this patch make everything better?] being maddeningly vague.

**Review by spocke — APPROVED** [review]

File: modules/ROOT/pages/autocorrect.adoc
Comment (brian-forte) [reply modules/ROOT/pages/autocorrect.adoc]: Both one-liners now edited.

FYI: _interferes with_ is maddeningly vague to me. But I’ve no info as to the nature of the interference.

File: modules/ROOT/partials/misc/autocorrect-options-notes.adoc
Comment (brian-forte) [reply modules/ROOT/partials/misc/autocorrect-options-notes.adoc]: The far more accurate NOTE admonition re interaction between Spell Checker Pro and Autocorrect is now a partial in its own right.

And `include` statements grabbing said partial are now in `autocorrect_autocorrect.adoc`, `autocorrect_capitalization.adoc`, *and* `autocorrect.adoc`.

File: modules/ROOT/pages/autocorrect.adoc
Comment (brian-forte) [reply modules/ROOT/pages/autocorrect.adoc]: The far more accurate NOTE admonition re interaction between Spell Checker Pro and Autocorrect is now a partial in its own right.

And `include` statements grabbing said partial are now in `autocorrect_autocorrect.adoc`, `autocorrect_capitalization.adoc`, *and* `autocorrect.adoc`.

File: modules/ROOT/partials/configuration/autocorrect_autocorrect.adoc
Comment (lnewson) [inline modules/ROOT/partials/configuration/autocorrect_autocorrect.adoc:8]: This change breaks the flow of information... the type, default and possible values no longer makes sense.

- [ ] Revert this change or come up with a different solution to prevent the mess with the flow of information

Comment (brian-forte) [reply modules/ROOT/partials/configuration/autocorrect_autocorrect.adoc]: I moved usage note to the end of both files.

This breaks the pattern of the Example being the last thing in these configuration documents. Which was my reason for the placement.

That said, putting the *Usage note* section at the end was my initial impulse. It’s an additional piece of information that makes sense only if the previous info is grokked.

File: modules/ROOT/pages/autocorrect.adoc
Comment (vpyshnenko) [inline modules/ROOT/pages/autocorrect.adoc:12]: need to add modules/ROOT/partials/misc/admon-requires-6.2v.adoc

**Review by vpyshnenko — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/autocorrect.adoc
Comment (tiny-ben-tran) [reply modules/ROOT/pages/autocorrect.adoc:12]: The current path is correct.

File: modules/ROOT/pages/autocorrect.adoc
Comment (MitchCTiny) [reply modules/ROOT/pages/autocorrect.adoc]: Good idea, I just added a bunch of words I felt were relevant when I made this branch.

File: modules/ROOT/pages/autocorrect.adoc
Comment (MitchCTiny) [reply modules/ROOT/pages/autocorrect.adoc]: I agree this should simply be `autocorrect`

File: modules/ROOT/pages/autocorrect.adoc
Comment (MitchCTiny) [inline modules/ROOT/pages/autocorrect.adoc:74]:
```suggestion
Auto-capitalization will not operate on the entered string if
+
  * a {productname} document contains any data other than a soft return character (Unicode U+2028, commonly entered by typing *Shift+Return*); and
  * the character immediately preceding an entered string is a soft return character.
+
```
I'm not sure what your preference is, just putting `If` on a single line followed by a list and then a final line looks funny to me.

**Review by MitchCTiny — APPROVED** [review]

**Review by lorenzo-pomili — APPROVED** [review]

File: modules/ROOT/pages/autocorrect.adoc
Comment (tiny-ben-tran) [reply modules/ROOT/pages/autocorrect.adoc:12]: @vpyshnenko, can you also please mark this request as addressed?

File: modules/ROOT/partials/misc/autocorrect-usage-notes.adoc
Comment (ephox-mogran) [inline modules/ROOT/partials/misc/autocorrect-usage-notes.adoc:5]: Both `Autocorrect` and `Capitalize` are turned off when changing the language (IIRC). This is just one of those annoying side-effects of us using `autocorrect` as a catch-all for all the corrections that occur, but having two different menu items (autocorrect and capitalize). I'm not sure if we need to clarify this.

**Review by ephox-mogran — APPROVED** [review]

File: modules/ROOT/partials/misc/autocorrect-usage-notes.adoc
Comment (MitchCTiny) [reply modules/ROOT/partials/misc/autocorrect-usage-notes.adoc:5]: Looking over this again, it seems that there aren't any references to the fact that `capitalization` is disabled by default with all languages. Also, the fact that this partial is included three times on the autocorrect page should be fixed.

File: modules/ROOT/pages/autocorrect.adoc
Comment (ephox-mogran) [inline modules/ROOT/pages/autocorrect.adoc:43]: (discussion): use backtick or something else to make it clear that this is intentional
```suggestion
For example, if `+cna+` (which the {pluginname} plugin treats as a mis-typing of *can*) is a desired string in a particular use-case, turning {pluginname} off is the currently recommended setting.
```

File: modules/ROOT/partials/misc/autocorrect-usage-notes.adoc
Comment (ephox-mogran) [inline modules/ROOT/partials/misc/autocorrect-usage-notes.adoc:1]: - [ ] remove the list formatting from this partial, and change the usages

File: modules/ROOT/pages/autocorrect.adoc
Comment (ephox-mogran) [inline modules/ROOT/pages/autocorrect.adoc:45]:
```suggestion
. End-users cannot, by default, turn autocorrect and capitalization on or off on hand-held devices.
```

File: modules/ROOT/partials/misc/autocorrect-usage-notes.adoc
Comment (ephox-mogran) [inline modules/ROOT/partials/misc/autocorrect-usage-notes.adoc:3]:
```suggestion
When the {pluginname} is used with the xref:introduction-to-tiny-spellchecker.adoc[Spell Checker Pro] Premium plugin, autocorrection and capitalization are disabled when a non-English language is selected.
```

File: modules/ROOT/partials/misc/autocorrect-usage-notes.adoc
Comment (ephox-mogran) [reply modules/ROOT/partials/misc/autocorrect-usage-notes.adoc:5]:
```suggestion
Autocorrection and capitalization can, however, be re-enabled by selecting the *Tools > Autocorrect* menu item and the "Tools > Capitalization" menu item respectively.
```

Comment (ephox-mogran) [reply modules/ROOT/partials/misc/autocorrect-usage-notes.adoc]: (discussion): Removed the usage notes.

**Pull request merged**
